src/App.jsx

```diff
@@ -14,33 +15,88 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const ACTIVE_BUTTON_ABC = 'ABC';
+const ACTIVE_BUTTON_LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getSortedGoods(goods, { activeSort, isReversed }) {
+  const sortedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (activeSort) {
+    sortedGoods.sort((good1, good2) => {
+      switch (activeSort) {
+        case ACTIVE_BUTTON_ABC:
+          return good1.localeCompare(good2);
+        case ACTIVE_BUTTON_LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (isReversed) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+}
+
+export const App = () => {
+  const [activeSort, setActiveSort] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const sortGoods = getSortedGoods(goodsFromServer, { activeSort, isReversed });
+  const getResetButton = activeSort || isReversed;
+
+  const reset = () => {
+    setActiveSort('');
+    setIsReversed(false);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info ${activeSort !== ACTIVE_BUTTON_ABC && 'is-light'}`}
+          onClick={() => setActiveSort(ACTIVE_BUTTON_ABC)}
+        >
+          Sort alphabetically
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          type="button"
+          className={`button is-success ${activeSort !== ACTIVE_BUTTON_LENGTH && 'is-light'}`}
```

Comment:
it is better to use libraries like classnames or clsx for conditional classNames